superset/assets/src/dashboard/components/Dashboard.jsx

```diff
@@ -170,6 +174,21 @@ class Dashboard extends React.PureComponent {
         this.props.actions.removeSliceFromDashboard(removedChartId),
       );
     }
+
+    if (nextProps.dashboardState.modalSliceId != this.props.dashboardState.modalSliceId) {
+      if (nextProps.dashboardState.modalSliceId != -1) {
+        // update modal  chart  state renedered
+        this.updateModalProps(nextProps.dashboardState.modalSliceId, "rendered", true)
```

Comment:
if true/false value is dependent on nextProps.dashboardState.modalSliceId value in that case we can remove if/else and use this value directly in third param.

Reply:
done